tests/testthat/test-methods.R

```diff
@@ -157,3 +157,43 @@ test_that("as.data.table.list is _not_ dispatched directly", {
   lints <- lint(text = "a = 1", linters = assignment_linter())
   expect_identical(nrow(data.table::as.data.table(lints)), 1L)
 })
+
+local({
+  # avoid impact of CLI mark-up on strwrap output.
+  #   (testthat, or cli, already do so, but force it explicitly here for emphasis)
+  withr::local_options(c(cli.num_colors = 0L))
+  # force "default" print method even on GHA
+  withr::local_envvar(c(GITHUB_ACTIONS = NA))
+
+  test_linter <- make_linter_from_xpath("*[1]", lint_message = "The quick brown fox jumps over the lazy dog.")
+
+  lints <- lint(text = "a", linters = test_linter())
+  lint <- lints[[1L]]
+
+  wrapped_strings <- c(
```

Comment:
I had to open my IDE to check these tests locally because I found it so hard to _see_ if the output indeed looked as expected. And this is making me wonder if we have finally found a place where we should use snapshot tests? WDYT?

Reply:
I could get the snapshot tests to work locally, and it seems to work, except for `format()`. 
If I do `cat(format(...))`, I get the following error:

```r
── Error (test-methods.R:183:7): format.lint, format.lints, print.lint, print.lints support optional message wrapping 80 ──
Error in `trace_back(top = getOption("testthat_topenv"), bottom = trace_env)`: Can't find `bottom` on the call tree.
Backtrace:
    ▆
 1. ├─rlang::eval_tidy(code, args)
 2. └─testthat::expect_snapshot(cat(format(lint, width = 20L))) at test-methods.R:183:7
 3.   └─testthat:::expect_snapshot_helper(...)
 4.     └─testthat::expect(...)
 5.       └─rlang::trace_back(top = getOption("testthat_topenv"), bottom = trace_env)
 6.         └─rlang:::trace_find_bottom(bottom, frames)
 7.           └─rlang::abort(msg, call = call)
```

If I don't `cat()`, then, of course, the snapshotted output isn't any better than just having the wrapped strings.

I can investigate this further after work, unless you already have some ideas why this might not be working.

Reply:
Seems like a bug (not sure if {testthat} or maybe even {rlang}).

It _does_ seem less appropriate to test the `format()` output with snapshots. But understand it's a bit awkward to mix snapshots with ugly formatted string outputs.

Easiest way to proceed would seem to be (1) file an upstream issue while we wait to figure out why snapshots won't work (2) leave a TODO here (3) merge with the working (but hard-to-read) tests we have already

Reply:
I'd prefer to go for the third option: merge as is and then create a follow-up issue (I can take care of this) for using snapshot tests.

I will approve this, but let's wait for @AshesITR to also resolve his comment before we merge.

Reply:
Done: https://github.com/r-lib/lintr/issues/2370